lib/openstack/amqp/openstack_qpid_event_monitor.rb

```diff
@@ -13,10 +13,25 @@ def self.available?(options = {})
     OpenstackQpidConnection.available? && test_connection(options)
   end
 
+  def self.connection_parameters(options = {})
+    username = options[:username] if options.key? :username
+    password = options[:password] if options.key? :password
+    [
+      options[:hostname],
+      options[:port],
+      username,
+      password
+    ].compact
```

Comment:
That's true, but originally `hostname` and `port` were both required.  So, this holds that assumption.  I guess I need to move the exception here to deal with missing hostname or port.

I was compacting to deal with the possible missing `username` or `password`, which are actually optional.